src/test/java/com/codeus/winter/config/PackageBeanRegistrationTest.java

```diff
@@ -30,65 +27,59 @@ class PackageBeanRegistrationTest {
     private PackageBeanRegistration beanRegistration;
 
     /**
-     * Sets up the test environment by mocking dependencies and initializing
-     * the {@link PackageBeanRegistration} instance.
+     * Sets up the test environment by mocking dependencies and initializing the
+     * {@link PackageBeanRegistration} instance.
      */
     @BeforeEach
     void setUp() {
-        packageScanner = mock(PackageScanner.class);
-        registry = mock(BeanDefinitionRegistry.class);
+        packageScanner = new PackageScannerImpl();
+        registry = new BeanDefinitionRegistryImpl();
         beanRegistration = new PackageBeanRegistration(packageScanner, registry);
     }
 
-    /**
-     * Tests that classes annotated with {@link Component} are successfully
-     * registered as {@link BeanDefinition}s in the registry.
-     */
     @Test
     void shouldRegisterBeansSuccessfully() {
-        Class<?> testClass = WinterComponent.class;
-        when(packageScanner.findClassesWithAnnotations(anyString(), anySet()))
-                .thenReturn(Set.of(testClass));
-        when(registry.containsBeanDefinition("winterComponent")).thenReturn(false);
-        beanRegistration.registerBeans("com.framework");
+        beanRegistration.registerBeans("com.codeus.winter.config");
 
-        verify(registry).registerBeanDefinition(eq("winterComponent"), any(BeanDefinitionImpl.class));
+        BeanDefinition definition = registry.getBeanDefinition("winterComponent");
+        assertNotNull(definition);
+        // bean name
+        assertEquals("com.codeus.winter.config.WinterComponent", definition.getBeanClassName());
+        // scope
+        assertEquals(SCOPE_PROTOTYPE, definition.getScope());
+        assertFalse(definition.isSingleton());
+        // depends on
+        String[] dependencies = {"com.codeus.winter.config.QualifierComponent",
+            "com.codeus.winter.config.AutowiredComponent"};
+        assertTrue(Arrays.equals(dependencies, definition.getDependsOn()));
```

Comment:
Can be simplified by using `assertArrayEquals`.

Reply:
added